libs/queue/src/main.sw

```diff
@@ -0,0 +1,98 @@
+// The `Queue` type corresponds to the same data structure.
+library;
+
+use std::vec::Vec;
+
+pub struct Queue<T> {
+    vec: Vec<T>,
+}
+
+impl<T> Queue<T> {
+    /// Create a new `Queue`
+    pub fn new() -> Self {
+        Self {
+            vec: Vec::new(),
+        }
+    }
+
+    /// Checks a `Queue` for emptiness
+    pub fn is_empty(self) -> bool {
+        self.vec.is_empty()
+    }
+
+    /// Gets the number of elements in the `Queue`
+    pub fn len(self) -> u64 {
+        self.vec.len()
+    }
+
+    /// Enqueues a `Queue`
+    pub fn enqueue(ref mut self, item: T) {
+        self.vec.push(item);
+    }
+
+    /// Dequeues a `Queue`
+    pub fn dequeue(ref mut self) -> Option<T> {
+        if self.vec.is_empty() {
+            Option::None
+        } else {
+            Option::Some(self.vec.remove(0))
+        }
+    }
+
+    /// Gets the last element
+    pub fn peek(self) -> Option<T> {
+        self.vec.get(0)
+    }
+}
+
+#[test()]
+fn test_new_queue() {
+    let new_queue: Queue<u64> = Queue::new();
+    assert(new_queue.is_empty() == true);
+    assert(new_queue.len() == 0);
+}
+
+#[test()]
+fn test_enqueue() {
+    let mut queue_for_enqueue: Queue<u64> = Queue::new();
+    queue_for_enqueue.enqueue(1);
+    queue_for_enqueue.enqueue(2);
+    queue_for_enqueue.enqueue(3);
+    assert(queue_for_enqueue.len() == 3);
```

Comment:
I would `enqueue` twice instead of 3 times too as 3 is unnecessary and after each addition I would assert that the length has increased by 1 otherwise this test proves that if you add 3 times then the length is 3 rather than after each addition the length increases by 1 because an item has been added.

Reply:
Done